charmcraft/application/commands/base.py

```diff
@@ -26,6 +29,14 @@ class CharmcraftCommand(craft_application.commands.AppCommand):
     format_option: bool = False
     _services: services.CharmcraftServiceFactory
 
+    def __init__(self, config: dict[str, Any] | None) -> None:
+        super().__init__(config)
+
+        if config is None:
```

Comment:
when is config `None`?

Reply:
I guess no `charmcraft.yaml`?

Reply:
Not sure TBH, but the craft-cli base accepts a `None` so I presumed it was for getting help or something.